File: net/Link.Message.Client/content/TextMessageContent.cs
using Link.Message.Client.utils;
using Newtonsoft.Json;

namespace Link.Message.Client.content
{
	/// <summary>
    /// 简单文本消息内容
	/// </summary>
	[JsonConverter(typeof(StringOrObjectConverter))]
	public class TextMessageContent : MessageContent, IStringOrObjectSerializable
	{
		public string Text { set; get; }

		public TextMessageContent(string text)
		{
			Guard.GuardReqiredString(text, "TextMessageContent's content is required.");

			Text = text;
			Type = MessageType.Text;
		}

	    public override string ToString()
	    {
	        return Text;
	    }
	}

}

File: java/link-message-client/src/main/java/link/message/client/content/TextMessageContent.java
package link.message.client.content;

import link.message.client.MessageType;
import link.message.client.StringOrObjectSerializable;
import link.message.client.utils.Guard;

/**
 * 简单文本消息内容
 * @author zhongt
 *
 */
public class TextMessageContent extends MessageContent implements StringOrObjectSerializable {
	private String text;
	
	public TextMessageContent(String text) {
		Guard.guardReqiredString(text, "TextMessageContent's content is required.");
		
		this.text = text;
		this.type = MessageType.TEXT.value();
	}

	@Override
	public String toString() {
		return text;
	}
}
